xmtp-inbox-ios/Views/ConversationListView.swift

```diff
@@ -18,7 +18,9 @@ struct ConversationListView: View {
 
 	@State private var mostRecentMessages = [String: DecodedMessage]()
 	@State private var status: LoadingStatus = .success
+	@State var isShowingNewMessage = false
```

Comment:
One thing I like to do is maybe add a `SheetButton` component instead of having to add a bunch of bool properties to views. That way the parent views don't need to worry so much about what sheets their include. Something like:

```swift
import SwiftUI

struct SheetButton<ButtonContent: View, SheetContent: View>: View {
	@State private var isPresented = false
	var buttonContent: () -> ButtonContent
	var sheetContent: () -> SheetContent

	var body: some View {
		Button(action: {
			self.isPresented.toggle()
		}) {
			buttonContent()
		}.sheet(isPresented: $isPresented) {
			sheetContent()
		}
	}
}

struct SheetButton_Previews: PreviewProvider {
	static var previews: some View {
		SheetButton {
			Text("Press me")
		}
		sheetContent: {
			Text("I'm a sheet!")
		}
	}
}
```

You don't need to do it in this PR, but it could be worth looking at soon.

Reply:
Ok I promise I gave this a shot, but I couldn't get it to play nicely with the FloatingButton which also takes in an `action`. Essentially it just showed a button that didn't tap into anything.